packages/mui/src/hooks/useDataGrid/index.ts

```diff
@@ -45,41 +47,50 @@ type DataGridPropsType = Required<
     | "disableRowSelectionOnClick"
     | "onStateChange"
     | "paginationMode"
+    | "processRowUpdate"
   >
 > &
   Pick<
     DataGridProps,
     "paginationModel" | "onPaginationModelChange" | "filterModel"
```

Comment:
```suggestion
  >
> &
  Pick<
    DataGridProps,
    "paginationModel" | "onPaginationModelChange" | "filterModel" | "processRowUpdate"
```

Let's move this here, I don't think this needs to be present for all `useDataGrid` usages. Placing it here will mark it as optional.

I think we can add `editable: boolean` prop to `useDataGrid` and make a conditional return for `processRowUpdate`. I'm not sure but sending this even without any editable columns may cause issues with current users implemented editable mode in their codebase.